mytardis_rocrate_builder/rocrate_builder.py

```diff
@@ -56,71 +76,107 @@ def __init__(
         self.crate = crate
         self.flatten_additional_properties = flatten_additional_properties
 
-    def _add_metadata_to_crate(
-        self, metadata_obj: MTMetadata, metadata_id: str, parent_id: IdentiferType
-    ) -> ContextEntity | None:
-        """Add a MyTardis Metadata object to the crate
+    def dereference_or_add(
```

Comment:
This seems a little bit of a strange mix of tasks, both dereferencing and adding a new object. Is this inherited behaviour or should we think about separating these functions out to make the separation of concerns more explicit?